model/task/task.go

```diff
@@ -4037,3 +4039,100 @@ func (t *Task) FindAbortingAndResettingDependencies() ([]Task, error) {
 	})
 	return FindAll(q)
 }
+
+// SortingValueBreakdown is the full breakdown of the final value used to sort on in the queue,
+// with accompanying breakdowns of priority and rank value.
+type SortingValueBreakdown struct {
+	TaskGroupLength    int64
+	TotalValue         int64
+	PriorityBreakdown  PriorityBreakdown
+	RankValueBreakdown RankValueBreakdown
+}
+
+// PriorityBreakdown contains information on how much various factors impacted the custom
+// priority value that is used in the queue sorting value equation.
+type PriorityBreakdown struct {
+	// InitialPriorityImpact represents how much of the total priority can be attributed to the
+	// original priority set on a task by a user.
+	InitialPriorityImpact int64
+	// TaskGroupImpact represents how much of the total priority can be attributed to a
+	// task being in a task group.
+	TaskGroupImpact int64
+	// GenerateTaskImpact represents how much of the total priority can be attributed to a
+	// task having a generate.tasks command in it.
+	GenerateTaskImpact int64
```

Comment:
extreme nit -- i think it might be more clear to call this GeneratorTaskImpact -- internally I think we usually refer to the generator task that way, and the generated tasks as generated, so generateTask is introducing a new vocabulary.